pkg/spoke/submarineragent/config_controller.go

```diff
@@ -573,3 +573,11 @@ func successCondition(gatewayNames []string) metav1.Condition {
 			strings.Join(gatewayNames, ",")),
 	}
 }
+
```

Comment:
The comment below was marked as resolved and I'm not getting option to unmark it. One follow up question.

What about the default case? Do we not have to distinguish between the clouds where we have automation (like AWS/GCP/RHOS/Azure) vs clouds like VmWare/BareMetal etc which requires manual preparation?

Reply:
Yes but only here - once AWS is moved here then the hub controller won't do any cloud prepare.

Reply:
My question is not about Hub/Spoke, but its about - dont we want to distinguish between clouds which have automation (like AWS/GCP/Azure/RHOS) vs the remaining?

Reply:
The isSpokePrepared API was exactly the same in the initial PR (so i added a comment to de-duplicate it) and was later modified in subsequent PRs. It looks fine now.